beacon-chain/sync/initial-sync/service.go

```diff
@@ -117,23 +117,12 @@ func (s *InitialSync) Start() {
 			panic(err)
 		}
 
-		// Read status code.
-		code, errMsg, err := sync.ReadStatusCode(strm, s.p2p.Encoding())
-		if err != nil {
-			panic(err)
-		}
-		if code != 0 {
-			log.Errorf("Request failed. Request was %+v", req)
-			panic(errMsg)
-		}
-
-		resp := make([]*eth.BeaconBlock, 0)
-		if err := s.p2p.Encoding().DecodeWithLength(strm, &resp); err != nil {
-			log.Error(err)
-			continue
-		}
-
-		for _, blk := range resp {
+		for i := req.StartSlot; i < req.StartSlot+(req.Count*req.Step); i += req.Step {
```

Comment:
Isn't this simpler to write like this?
```go
for i := 0; i < req.Count; i++ {
```

Reply:
What happens if there is a skip slot or the server returns less than the expected count?